### PR TITLE
Version requirements fixes

### PR DESCRIPTION
**Pull request by will-moore**
Work in progress...

In removing the "Components" section, I noticed some differences between that section and the other tables:

 - Postgres 9.6 on OMERO 6.0. Deprecated or Dropped?
 - Postgrex 12 on OMERO 5.6 Upcoming or Supported?
 - Python 3.5 on OMERO 5.6 Supported or Unsupported?
 -

### CODE REVIEW

**Comment by joshmoore** [comment]
My vote for 5.6 column
 - postgres 9.6: I had added [deprecated](https://github.com/ome/ome-documentation/pull/2034/files#diff-280d82ddeef248b5173afcfd2b6a62bbR160) but I can see the argument for "dropped"
 - postgres 12: I'd leave it as "upcoming". I don't know of any testing currently ongoing and it's pretty straight-forward to _add_ support later.
 - python 3.5: in this case, I'd roll back to "upcoming" , i.e. not yet supported but we could imagine adding it.

**Comment by joshmoore** [comment]
I like the new table at the top. If it could be a bit more horizontal that'd be nice. Separately, I could imagine making the "upcoming" arrow red or at least yellow/orange.

**Comment by jburel** [comment]
as discussed before py3.5 is the "default" on debian 9 and ubuntu16.04. I split the list on the guides  so they are not marked as recommended 
pg12 is the "default" on ubuntu 18.04 if I remember correctly

**Comment by joshmoore** [comment]
> as discussed before py3.5 is the "default" on debian 9 and ubuntu16.04

Sure, but in general I think we should be conservative what we specify here for December and then open it up as we go.

> pg12 is the "default" on ubuntu 18.04

`apt-get install postgresql` installs 10 in a docker container.

**Comment by jburel** [comment]
my comment about 3.5 is that I agree that we should change the classification. If we go for upcoming in the requirements I will use the same term in the server-installation page. Currently it is supported
pg: I got lost with the various versions on the various os I have been testing :-)

**Comment by will-moore** [comment]
Another mismatch between the Components tables and nginx section:
 - nginx 1.16 on OMERO 5.6 is Upcoming or Recommended (or Supported)?

**Comment by manics** [comment]
How about we ditch the detailed Nginx versions and just say 1.14 or newer is recommended?

File: omero/sysadmins/version-requirements.rst [hunk 1]
```diff
@@ -22,318 +22,83 @@ plan ahead and ensure that prerequisites are in place ahead of time to
...
+      - |Recommended|
+      - |Supported|
+    * - 1.14
+      - |Upcoming|
+      - |Supported|
+      - |Recommended|
+    * - 1.16
+      - |Upcoming|
+      - |Upcoming|
+      - |Upcoming|
```
Comment (manics) [inline omero/sysadmins/version-requirements.rst]:
```suggestion
      - |Recommended|
```

**Comment by pwalczysko** [comment]
```
Criteria for what is considered to be supportable includes whether
support by both upstream developers and operating system distributions
will be available for the lifetime of the |version| release (including
security support), and also upon our resources allocated to CI and
testing.
```


https://github.com/ome/ome-documentation/pull/2041/files#diff-280d82ddeef248b5173afcfd2b6a62bbR10 - rather clanky sentence. Maybe move the bracket ``(including security...)`` just after the words ``...whether support`` - and maybe remove the bracket.

File: omero/sysadmins/version-requirements.rst [hunk 2]
```diff
@@ -445,6 +210,19 @@ CI and testing. Only the last two LTS releases are supported (being a
...
+-----------------
+
+Client support only.
+See `blog post explanation <https://blog.openmicroscopy.org/tech-issues/future-plans/deployment/2016/03/22/windows-support/>`_
+
+MacOS X
+-------
+
+MacOS X is typically suited only to client use, not serious server
+deployment, although the server can be expected to build on versions with
```
Comment (manics) [inline omero/sysadmins/version-requirements.rst]:
```suggestion
deployment, although the server can be expected to run on versions with
```

File: omero/sysadmins/version-requirements.rst [hunk 3]
```diff
@@ -445,6 +210,19 @@ CI and testing. Only the last two LTS releases are supported (being a
...
+current security support for development purposes.
```
Comment (manics) [inline omero/sysadmins/version-requirements.rst]:
```suggestion
current security support for testing purposes.
```

**Comment by pwalczysko** [comment]
```
This section contains a summary of the changes made to the minimum
version requirements for the current (|version|) release and also
**possible** changes for the next release, albeit **tentatively at this
point**.
```

What is done ``tentatively`` here ? Possibly the ``possible changes``. This could be expressed better, again, very long sentence, was pondering what is the ``tentative`` point here.

**Comment by pwalczysko** [comment]
```
The following sections provide more detailed information and the rationale for the changes.
```

Drop ``more``

File: omero/sysadmins/version-requirements.rst [hunk 4]
```diff
@@ -979,7 +776,7 @@ OMERO support policies
       - TBA
       - |Upcoming|
       - |Upcoming|
-      - |Recommended|
+      - |Supported|
```
Comment (manics) [inline omero/sysadmins/version-requirements.rst]:
```suggestion
      - |Recommended|
```

**Review by manics — COMMENTED** [review]
The Django section needs to be updated to 1.11:
> The supported version of the Django module used by OMERO.web (1.8) requires Python 2.7. The older version (1.6) will work with Python 2.6 but lacks security support so should not be used.

**Comment by pwalczysko** [comment]
```
The intent is to provide a roadmap in order that sysadmins may plan ahead and ensure that prerequisites are in place ahead of time to ease future upgrades. 
```

Drop ``ahead of time to ease`` and replace it with ``for``

**Comment by pwalczysko** [comment]
Screen Shot 2019-12-04 at 10 28 39
the formatting of ``for ice ....`` seems raw

**Comment by pwalczysko** [comment]
Screen Shot 2019-12-04 at 10 43 45

Very misformatted - hard to read

**Comment by pwalczysko** [comment]
Screen Shot 2019-12-04 at 10 50 59

Is Homebrew an operating system ?

**Comment by pwalczysko** [comment]
Screen Shot 2019-12-04 at 10 53 13

The "details" at the bottom of the Postgres table - misformatted, and it points to just the same page as the "Overview" link above the same table - remove this line - just confusing

**Comment by pwalczysko** [comment]
Screen Shot 2019-12-04 at 10 56 51
Screen Shot 2019-12-04 at 10 56 44

This misformatting in Python table makes it hard to understand what is referenced.

**Comment by pwalczysko** [comment]
```
The limiting factor will be distribution support for 2.7 as major packages are slowly switching to 3.x and this might cause problems if our python module dependencies are no longer available without major effort.
```
Leave out the ``...if our python.......effort.`` part.

**Comment by pwalczysko** [comment]
```
The supported version of the Django module used by OMERO.web (1.8) requires Python 2.7. The older version (1.6) will work with Python 2.6 but lacks security support so should not be used.
```

Is this true ? I mean, do we use Django which requires Python 2.7 and this is supported ? If not, remove the sentence.

**Comment by pwalczysko** [comment]
Screen Shot 2019-12-04 at 11 09 22

In the Ice section, the sentence
``If no version is provided, a suitable repository is indicated.``
does not seem to be accurate - cannot see any suitable repo indicated in the table - remove it.

**Comment by pwalczysko** [comment]
Finished testing.

File: omero/sysadmins/version-requirements.rst [hunk 5]
```diff
@@ -882,6 +663,22 @@ OMERO support policies
       - |Recommended|
       - |Recommended|
       - `Reference <https://access.redhat.com/articles/1299013>`__
+    * - 12
+      - from Sep 2018
+      - to Oct 2024
+      - |Unsupported|
+      - |Unsupported|
+      - |Supported|
+      - |Recommended|
```
Comment (manics) [inline omero/sysadmins/version-requirements.rst:655]:
```suggestion
      - |Supported|
```
12 isn't an LTS: https://www.oracle.com/technetwork/java/java-se-support-roadmap.html

Comment (joshmoore) [reply omero/sysadmins/version-requirements.rst:655]: Neither is 13. That's why I omitted them. Perhaps a different type of "unsupported/" could cover them.

**Comment by mtbc** [comment]
Staged at https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-docs/lastSuccessfulBuild/artifact/omero/_build/html/sysadmins/version-requirements.html.

File: omero/sysadmins/version-requirements.rst (hunk 5 above)
Comment (manics) [reply omero/sysadmins/version-requirements.rst:655]: Maybe change this to `Unsupported` then? We don't need to go into details, and we're also not saying it's broken.

**Review by pwalczysko — APPROVED** [review]

**Comment by mtbc** [comment]
It's okay that OMERO 5.6 has two different version of NGINX as recommended / optimal?

I guess we do the same for Java for OMERO 6.0 which the first NGINX table doesn't mention.

**Comment by mtbc** [comment]
Might should list which distributions carry later PostgreSQL if that's what we're recommending.

**Comment by mtbc** [comment]
(None of the above blocks this PR.)

**Pull request merged**